src/Servers/IIS/test/Common.FunctionalTests/Inprocess/StartupTests.cs

```diff
@@ -107,6 +108,77 @@ public async Task StartsWithDotnetOnThePath(string path)
             Assert.Equal(1, TestSink.Writes.Count(w => w.Message.Contains("Invoking where.exe to find dotnet.exe")));
         }
 
+        [ConditionalTheory]
+        [InlineData(RuntimeArchitecture.x64)]
+        [InlineData(RuntimeArchitecture.x86)]
+        [RequiresIIS(IISCapability.PoolEnvironmentVariables)]
+        public async Task StartsWithDotnetInstallLocation(RuntimeArchitecture runtimeArchitecture)
+        {
+            var deploymentParameters = _fixture.GetBaseDeploymentParameters(publish: true);
+            deploymentParameters.RuntimeArchitecture = runtimeArchitecture;
+
+            deploymentParameters.EnvironmentVariables["PATH"] = "";
+            deploymentParameters.WebConfigActionList.Add(WebConfigHelpers.AddOrModifyAspNetCoreSection("processPath", "dotnet"));
+
+            // Key is always in 32bit view
+            using (var localMachine = RegistryKey.OpenBaseKey(RegistryHive.LocalMachine, RegistryView.Registry32))
```

Comment:
Why is it always in 32 bit view?

Reply:
That's the design. Installer always sets 32bit registry key.